scripts/tests/test_zkai_attention_derived_d128_snark_statement_receipt_gate.py

```
@@ -0,0 +1,181 @@
+        self.assertFalse(check["mutated_accepted"])
+        self.assertEqual(check["rejection_layer"], "external_proof_verifier")
+
+    def test_metric_smuggling_and_unknown_fields_fail_closed(self) -> None:
+        for name in (
+            "proof_size_metric_smuggled",
+            "verifier_time_metric_smuggled",
+            "proof_generation_time_metric_smuggled",
+            "unknown_top_level_field_added",
+        ):
+            _surface, receipt = GATE.mutated_receipts()[name]
+            with self.assertRaises(GATE.AttentionDerivedD128SnarkReceiptError):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)